### DISCUSSION TITLE
Removing JS runtime entirely

### DISCUSSION THREAD

**Discussion by Befo** [wailsapp/wails#2025] [Ideas]
Hello!
I am approaching go from the JS/TS world and this project really got my attention. I think this is the right direction for app development and with wasm it has insane potential imo. Looking there I was wondering if you see feasible removing the JS runtime from the webview renderer and to have Go code (either full fledged framework or not) doing whatever JS is doing in the webview lib, either still leveraging existing but modified webview engines or having a brand new one. Completely replacing JS with Go would allow companies to build wasm apps only using Go and HTML + CSS. I would love to hear what you think about it and I bet some of you thought about this.

Cheers and keep up the good work!!

**Comment by leaanthony** [comment]
Hi 👋 That's certainly something that's being talked about right now. If you have any experience or insights in doing this, please add them to this thread 👍

**Reply by Befo** [reply]
Ok! Unfortunately I do not have any at the moment, I just have in my mind that it would be cool to have an HTML/CSS based cross-platform UI toolkit like qt or gtk are. I still need to compare it, but I like also to find a challenging Go project to dive deeper into the language. I think I will try to fork the blink or webkit engines hoping that would allow the lib to stay up to date with web evolutions. If I get myself to do anything and I find something useful I will report it here. I was just curious to know what kind of problems did you encounter that prevented you to do so, but I assume that the main one is that it is too daunting to keep up while developing Wails at the same time.

**Reply by leaanthony** [reply]
@SamHennessy

**Reply by SamHennessy** [reply]
Hi @Befo, you mentioned WASM as a possible solution, but I was wondering if WASM is what you want or if it is not having to code in JavaScript. Or something else?

**Reply by Befo** [reply]
Hello @SamHennessy, it is wasm I am aiming at while having a ui toolkit built around HTML and CSS without JS. I realised after that probably what I want is more of a HTML and CSS renderer that does not uses JS as its interactivity language. It's just an idea, I am just investigating if would be useful or not.

**Reply by leaanthony** [reply]
I think there are already projects that do this (go -> wasm with drawing).
